Matrix.SynapseInterop.Database/SynapseDbContext.cs

```diff
@@ -18,5 +31,52 @@ protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
         {
             optionsBuilder.UseNpgsql(_connString);
         }
+
+        public List<DeviceContentSet> GetNewDevicesForDestination(string destination, int limit)
+        {
+            List<DeviceContentSet> set = new List<DeviceContentSet>();
+            foreach (var devicePoke in DeviceListsOutboundPokes
+                .Where(poke => poke.Destination == destination && poke.Sent == false)
+                .Take(limit)
+                .OrderBy(poke => poke.StreamId).ToList())
+            {
+                DeviceContentSet contentSet = new DeviceContentSet
+                {
+                    device_id = devicePoke.DeviceId,
+                    stream_id = devicePoke.StreamId,
+                    user_id = devicePoke.UserId,
+                    deleted = false,
+                };
+                var json = E2EDeviceKeysJson
+                    .SingleOrDefault(devKeys => devKeys.DeviceId == devicePoke.DeviceId && devKeys.UserId == devicePoke.UserId);
+                contentSet.keys = JObject.Parse(json.KeyJson);
+                contentSet.device_display_name = Devices.SingleOrDefault(dev =>
+                    dev.UserId == devicePoke.UserId && dev.DeviceId == devicePoke.DeviceId)
+                    ?.DisplayName;
+                set.Add(contentSet);
+            }
+            return set;
+        }
+
+        public List<EventJsonSet> GetAllNewEventsStream(int fromId, int currentId, int limit)
```

Comment:
lint that doesn't exist: `IEnumberable` should be exposed, not a mutable collection